transformer_engine/common/fused_attn/fused_attn_f16_arbitrary_seqlen.cu

```diff
@@ -145,22 +145,22 @@ void fused_attn_arbitrary_seqlen_fwd_impl(
                                        .set_name("offset_q")
                                        .set_dim({b + 1, 1, 1, 1})
                                        .set_stride({1, 1, 1, 1})
-                                       .set_data_type(fe::DataType_t::INT32));
+                                       .set_data_type(fe::DataType_t::INT64));
```

Comment:
Is this compatible with old cuDNN? If not, we need a variable for different cuDNN runtime version.

```
auto offset_type = fe::DataType_t::INT64;
if (cudnn run time version < ???) { offset_type = fe::DataType_t::INT32; }
```

Reply:
Asked the cudnn folks and indeed this feature is not in cuDNN <9.5 so I will refactor to check. We will also need to add some more logic to warn the user if they use configs that require too large of size.